test/e2e/framework.go

```diff
@@ -44,13 +44,23 @@ type Framework struct {
 	// It will read the data every 30 seconds from all Nodes and print summary during afterEach.
 	GatherKubeSystemResourceUsageData bool
 	gatherer                          containerResourceGatherer
+
+	// Allows to override the initialization of the namespace
+	nsCreateFunc func(string, *client.Client) (*api.Namespace, error)
 }
 
 // NewFramework makes a new framework and sets up a BeforeEach/AfterEach for
 // you (you can write additional before/after each functions).
 func NewFramework(baseName string) *Framework {
+	return InitializeFramework(baseName, createTestingNS)
+}
+
+// InitializeFramework initialize the framework by allowing to pass a custom
+// namespace creation function.
+func InitializeFramework(baseName string, nsCreateFunc func(string, *client.Client) (*api.Namespace, error)) *Framework {
```

Comment:
sorry to rework this again, but given the number of additional things added to the `Framework` struct since this PR was first written (NamespaceDeletionTimeout, GatherKubeSystemResourceUsageData, etc) I'm wondering if maybe you should just override `nsCreateFunc` when necessary, rather than having a separate initialization method.